sdk/tables/azure-data-tables/azure/data/tables/_base_client.py

```diff
@@ -33,74 +28,78 @@
     DistributedTracingPolicy,
     HttpLoggingPolicy,
     UserAgentPolicy,
-    AzureSasCredentialPolicy
+    AzureSasCredentialPolicy,
+    NetworkTraceLoggingPolicy,
+    CustomHookPolicy,
+    RequestIdPolicy
 )
 
-from ._common_conversion import _to_utc_datetime, _is_cosmos_endpoint
+from ._generated import AzureTable
+from ._common_conversion import _is_cosmos_endpoint
 from ._shared_access_signature import QueryStringConstants
 from ._constants import (
     STORAGE_OAUTH_SCOPE,
     SERVICE_HOST_BASE,
-    CONNECTION_TIMEOUT,
-    READ_TIMEOUT,
 )
 from ._models import LocationMode, BatchTransactionResult
 from ._authentication import SharedKeyCredentialPolicy
 from ._policies import (
     CosmosPatchTransformPolicy,
     StorageHeadersPolicy,
-    StorageContentValidation,
-    StorageRequestHook,
-    StorageResponseHook,
-    StorageLoggingPolicy,
     StorageHosts,
     TablesRetryPolicy,
 )
 from ._models import BatchErrorException
 from ._sdk_moniker import SDK_MONIKER
 
-if TYPE_CHECKING:
-    from typing import (  # pylint: disable=ungrouped-imports
-        Union,
-        Optional,
-        Any,
-        Iterable,
-        Dict,
-        List,
-        Type,
-        Tuple,
-    )
-
-_LOGGER = logging.getLogger(__name__)
-_SERVICE_PARAMS = {
-    "blob": {"primary": "BlobEndpoint", "secondary": "BlobSecondaryEndpoint"},
-    "queue": {"primary": "QueueEndpoint", "secondary": "QueueSecondaryEndpoint"},
-    "file": {"primary": "FileEndpoint", "secondary": "FileSecondaryEndpoint"},
-    "dfs": {"primary": "BlobEndpoint", "secondary": "BlobEndpoint"},
-    "table": {"primary": "TableEndpoint", "secondary": "TableSecondaryEndpoint"},
-}
-
-
-class StorageAccountHostsMixin(object):  # pylint: disable=too-many-instance-attributes
+_SUPPORTED_API_VERSIONS = ["2019-02-02", "2019-07-07"]
+
+
+def get_api_version(kwargs, default):
+    # type: (Dict[str, Any], str) -> str
+    api_version = kwargs.pop("api_version", None)
+    if api_version and api_version not in _SUPPORTED_API_VERSIONS:
+        versions = "\n".join(_SUPPORTED_API_VERSIONS)
+        raise ValueError(
+            "Unsupported API version '{}'. Please select from:\n{}".format(
+                api_version, versions
+            )
+        )
+    return api_version or default
```

Comment:
One of the guidelines is that all SDKs support an API version parameter. So that part is mandatory. However in next stand up we should confirm what all languages are setting as the default API version.